### PR TITLE
Add TA sync code

### PR DESCRIPTION
**Pull request by rshipp**
### Description

Task: https://app.asana.com/0/1174274412967132/1199733407122981/f

When a TA exists in Salesforce, add them to the "Teaching Assistants" section of the "Fellow course" in Canvas, with a TA role, and _without_ the "limit interactions to people in your section" flag. 

Note: This code has not, and *can not* be tested, because no TA role exists in the test Salesforce server.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: config/initializers/constants.rb [hunk 1]
```diff
@@ -17,4 +17,5 @@ class RoleConstants
 # Sections
 class SectionConstants
```
Comment (jessouyangbraven) [inline config/initializers/constants.rb:18]: Not related to this PR: should this class also have cohort and cohort schedule in it?

**Review by jessouyangbraven — APPROVED** [review]
This looks reasonable to me.

File: app/services/sync_portal_enrollment_for_account.rb [hunk 2]
```diff
@@ -51,6 +51,9 @@ def add_enrollment!
     when SalesforceAPI::FELLOW
       sync_enrollment(sf_program.fellow_course_id, RoleConstants::STUDENT_ENROLLMENT,
                       course_section_name)
+    when SalesforceAPI::TEACHING_ASSISTANT
+      sync_enrollment(sf_program.fellow_course_id, RoleConstants::TA_ENROLLMENT,
+                      SectionConstants::TA_SECTION, limit_privileges_to_course_section=false)
```
Comment (sadleb) [inline app/services/sync_portal_enrollment_for_account.rb:56]: Should you add a spec checking that `limit_privileges_to_course_section=false` is passed in and comment about why? Aka. "Teaching Assistants are put in a generic Canvas section and need to be able to access and grade things across the who course. This allows them to grade folks in any section."

**Review by sadleb — APPROVED** [review]
One comment, but not merge blocking.

File: config/initializers/constants.rb (hunk 1 above)
Comment (rshipp) [reply config/initializers/constants.rb:18]: cohort and cohort schedule aren't constants, they're strings defined by salesforce fields. unless i'm misunderstanding the question

**Pull request merged**